uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/rules/UncaughtExceptionHandlerRule.java

```diff
@@ -0,0 +1,57 @@
+package com.microsoft.identity.client.ui.automation.rules;
+
+import android.util.Log;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * A Test Rule to set a default handler for uncaught exceptions from threads that may crash during automation.
+ */
+public class UncaughtExceptionHandlerRule implements TestRule {
+
+    private final static Queue<Throwable> caughtThrowables = new ConcurrentLinkedQueue<>();
```

Comment:
I think this doesn't need to be static as it is only accessed within this class. Can be changed to member variable